components/draw/PointOrPolygonMode.svelte

```diff
@@ -0,0 +1,79 @@
+<script>
+  import {
+    map,
+    gjScheme,
+    newFeatureId,
+    setCurrentlyEditing,
+  } from "../../stores.js";
+
+  // "point" or "polygon"
+  export let thisMode;
+  export let mode;
+  export let changeMode;
+  export let drawControls;
+
+  // This is state that needs to survive between draw.create and draw.modechange
+  let createdFeatureId;
+
+  export function start() {
+    drawControls.changeMode(`draw_${thisMode}`);
+  }
+  export function stop() {
+    createdFeatureId = null;
+    drawControls.changeMode("static");
+    drawControls.deleteAll();
+  }
+
+  $map.on("draw.create", (e) => {
+    if (mode == thisMode) {
+      // Assume there's exactly 1 feature
+      const feature = e.features[0];
+
+      gjScheme.update((gj) => {
+        feature.properties.intervention_type =
+          thisMode == "point" ? "other" : "area";
+        feature.id = newFeatureId(gj);
+        gj.features.push(feature);
+
+        // Remember this, so modechange below works
+        createdFeatureId = feature.id;
+
+        // It'd be great to do the callback below right now, but mapbox-gl-draw
+        // will infinitely recurse and blow up if we mutate it inside
+        // draw.create.
+
+        return gj;
+      });
+    }
+  });
+
+  $map.on("draw.modechange", (e) => {
```

Comment:
This handles behaviour immediately after a point is created or a Polygon is finished.